RailgunNet/System/Encoding/RailBitBuffer.cs

```diff
@@ -601,12 +601,95 @@ public string ReadString()
             uint length = Read(STRING_LENGTH_BITS);
             for (int i = 0; i < length; i++)
             {
-                builder.Append((char) Read(ASCII_BITS));
+                builder.Append((char)Read(ASCII_BITS));
             }
 
             return builder.ToString();
         }
         #endregion
+
+        #region UInt64
+        /// <summary>
+        ///     Writes using an elastic number of bytes based on number size:
+        ///     Bits   Min Dec    Max Dec     Max Hex     Bytes Used
+        ///     0-7    0          127         0x0000007F  1 byte
+        ///     8-14   128        1023        0x00003FFF  2 bytes
+        ///     15-21  1024       2097151     0x001FFFFF  3 bytes
+        ///     22-28  2097152    268435455   0x0FFFFFFF  4 bytes
+        ///     28-32  268435456  4294967295  0xFFFFFFFF  5 bytes
+        /// </summary>
+        [Encoder]
+        public void WriteUInt64(ulong val)
```

Comment:
Since its variable length anyways, you could just change the uint encoder & decoder to ulong. I don't see a reason to duplicating it.